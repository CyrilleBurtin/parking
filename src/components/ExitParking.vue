<script setup lang="ts">
import { ref } from 'vue';
import { useParkingLots } from '@/stores/parkingLots';
import { getAvailableLot } from '../utils/availableLots';

const { releaseLots, lots } = useParkingLots();
const ticketNumber = ref();
const ticketError = ref(false);

const returnTicket = () => {
  const availableLots = getAvailableLot(lots);
  if (ticketNumber.value < 0 || ticketNumber.value > 9) {
    ticketError.value = true;
  }

  if (!availableLots.includes(ticketNumber.value)) {
    releaseLots(ticketNumber.value);
    ticketNumber.value = null;
  } else {
    ticketError.value = true;
    ticketNumber.value = null;
  }
};
</script>

<template>
  <div>
    <label for="number-input" class="block mb-2 text-sm font-medium text-gray-900"
      >Entrez votre numéro de place pour sortir</label
    >
    <input
      type="number"
      v-model="ticketNumber"
      min="0"
      max="9"
      id="number-input"
      aria-describedby="helper-text-explanation"
      class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5"
      required
    />
    <p v-if="ticketError">
      "Le numéro de ticket entré correspond à une place déjà libre. Veuillez ressaisir votre numéro
      de place."
    </p>
    <button
      @click="returnTicket"
      :disabled="!ticketNumber"
      class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-4 rounded disabled:bg-slate-500"
    >
      exit parking
    </button>
  </div>
</template>
