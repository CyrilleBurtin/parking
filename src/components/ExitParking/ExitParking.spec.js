import { describe, it, expect, beforeEach } from 'vitest';
import { render } from '@testing-library/vue';
import ExitParking from './ExitParking.vue';
import { setActivePinia, createPinia } from 'pinia';

describe('ExitTicket', () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });
  it('shoulrender the component', () => {
    const { container } = render(ExitParking);

    expect(container).toMatchSnapshot();
  });
});
