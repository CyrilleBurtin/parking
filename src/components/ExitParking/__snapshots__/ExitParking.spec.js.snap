// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExitTicket > shoulrender the component 1`] = `
<div>
  <div
    class="max-w-sm p-6 bg-white border border-gray-200 rounded-lg shadow dark:bg-gray-800 dark:border-gray-700"
  >
    <a
      href="#"
    >
      <h5
        class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
      >
        
        <h2>
          Sortie du parking
        </h2>
        
      </h5>
    </a>
    <p
      class="mb-3 font-normal text-gray-700 dark:text-gray-400"
    >
      
      <p>
        Entrez votre num√©ro de place pour sortir
      </p>
      <input
        aria-describedby="helper-text-explanation"
        class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5"
        id="number-input"
        max="10"
        min="1"
        required=""
        type="number"
      />
      
    </p>
    
    <button
      class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-4 rounded disabled:bg-slate-500"
      disabled=""
    >
       exit parking 
    </button>
    
    <p
      class="text-green-500"
    >
      
      
    </p>
    <p
      class="text-red-500"
    >
      
      
    </p>
  </div>
</div>
`;
